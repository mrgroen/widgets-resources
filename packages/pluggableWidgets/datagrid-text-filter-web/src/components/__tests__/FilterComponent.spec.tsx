import { shallow } from "enzyme";
import { createElement } from "react";
import { FilterComponent } from "../FilterComponent";

jest.useFakeTimers();

describe("Filter component", () => {
    it("renders correctly", () => {
        const component = shallow(
            <FilterComponent adjustable defaultFilter="contains" delay={500} filterDispatcher={jest.fn()} />
        );

        expect(component).toMatchSnapshot();
    });

    it("renders correctly when not adjustable by user", () => {
        const component = shallow(
            <FilterComponent adjustable={false} defaultFilter="contains" delay={500} filterDispatcher={jest.fn()} />
        );

        expect(component).toMatchSnapshot();
    });

    it("renders correctly with aria labels", () => {
        const component = shallow(
            <FilterComponent
                adjustable
                screenReaderButtonCaption="my label"
                screenReaderInputCaption="my label"
                defaultFilter="contains"
                delay={500}
                filterDispatcher={jest.fn()}
            />
        );

        expect(component).toMatchSnapshot();
    });

    it("calls filterDispatcher when value changes", () => {
        const filterDispatcher = jest.fn();
        const component = shallow(
            <FilterComponent adjustable defaultFilter="contains" delay={500} filterDispatcher={filterDispatcher} />
        );

        const input = component.find("input");
        input.simulate("change", { target: { value: "test" } });

        expect(filterDispatcher).toBeCalled();
    });

    it("debounces calls for filterDispatcher when value changes", () => {
        const filterDispatcher = jest.fn();
        const component = shallow(
            <FilterComponent adjustable defaultFilter="contains" delay={500} filterDispatcher={filterDispatcher} />
        );

        // Initial call with default filter
        expect(filterDispatcher).toBeCalledTimes(1);

        const input = component.find("input");
        input.simulate("change", { target: { value: "test" } });
        jest.advanceTimersByTime(499);
        input.simulate("change", { target: { value: "test2" } });
        input.simulate("change", { target: { value: "test3" } });
        jest.advanceTimersByTime(500);

        expect(filterDispatcher).toBeCalledTimes(2);

        input.simulate("change", { target: { value: "test" } });
        jest.advanceTimersByTime(500);

        expect(filterDispatcher).toBeCalledTimes(3);
    });
});
