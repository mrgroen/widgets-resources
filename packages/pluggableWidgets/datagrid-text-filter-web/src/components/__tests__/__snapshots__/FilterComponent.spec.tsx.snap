// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter component renders correctly 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <FilterSelector
    defaultFilter="contains"
    onChange={[Function]}
  />
  <input
    className="form-control filter-input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Filter component renders correctly when not adjustable by user 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <input
    className="form-control"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;

exports[`Filter component renders correctly with aria labels 1`] = `
<div
  className="filter-container"
  data-focusindex={0}
>
  <FilterSelector
    ariaLabel="my label"
    defaultFilter="contains"
    onChange={[Function]}
  />
  <input
    aria-label="my label"
    className="form-control filter-input"
    onChange={[Function]}
    type="text"
    value=""
  />
</div>
`;
